ui/bottompanel.go

```diff
@@ -87,6 +90,9 @@ func NewBottomPanel(p *player.Player, contr *controller.Controller) *BottomPanel
 	bp.Controls.OnSeek(func(f float64) {
 		p.Seek(fmt.Sprintf("%d", int(f*100)), player.SeekAbsolutePercent)
 	})
+	bp.Controls.OnChangeLoopMode(func() {
+		p.SetNextLoopMode()
```

Comment:
I'd like to slightly change the approach of this PR to have the bottom panel communicate through the PlaybackManager rather than directly to the player. Eventually I plan to refactor for everything to flow through the PlaybackManager because: A) to support reordering the play queue, I'd like the MPV player to only have the current and next track in its queue, and only the PlaybackManager managing the full queue. B) Eventually there may be multiple player backends, eg the local MPV player, a DLNA backend, a Chromecast backend, and then the PlaybackManager would be in charge of relaying comands to the current player backend.

So I'd add SetNextLoopMode as a method on the PlaybackManager (the PlaybackManager's impl. can just call down to player.SetNextLoopMode), and have bottompanel interface through the playback manager. If you want to do more refactoring, you could get rid of BottomPanel.SetPlaybackManager, and just make the playbackManager a constructor argument alongside the Player. It was a mistake to make SetPlaybackManager a setter method, since the BottomPanel needs the PlaybackManager for the play time update anyway, and there will only ever be one playbackmanager

Reply:
I have made that refactor to reduce the direct dependency between BottomPanel and the Player! Feel free to check if that's what you were looking for.

> A) to support reordering the play queue, I'd like the MPV player to only have the current and next track in its queue, and only the PlaybackManager managing the full queue.

This will make the current loop logic to require a refactor soon, as it depends on MPV knowing the full queue. It will be interesting to open a GitHub discussion for that design change, as I was also thinking about relying more on MPV's shuffle functionality, something that could be opposite to what you are proposing here.